packages/@react-spectrum/s2/src/NotificationBadge.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps, DOMProps, DOMRef, DOMRefValue} from '@react-types/shared';
+import {ContextValue, SlotProps} from 'react-aria-components';
+import {filterDOMProps} from '@react-aria/utils';
+import {fontRelative, style} from '../style' with {type: 'macro'};
+import {getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {NumberFormatter} from '@internationalized/number';
+import React, {createContext, forwardRef} from 'react';
+import {useDOMRef} from '@react-spectrum/utils';
+import {useLocale, useLocalizedStringFormatter} from '@react-aria/i18n';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface NotificationBadgeStyleProps {
+  /**
+   * The size of the notification badge.
+   *
+   * @default 'S'
+   */
+  size?: 'S' | 'M' | 'L' | 'XL'
+}
+
+export interface NotificationBadgeProps extends DOMProps, AriaLabelingProps, StyleProps, NotificationBadgeStyleProps, SlotProps {
+  /**
+   * The value to be displayed in the notification badge.
+   */
+  value?: number
```

Comment:
`undefined` is a teeny bit weird, maybe it's better to make `value: number | null` so that it's explicitly empty? i could be persuaded either way. thoughts from the team?

Reply:
hm im fine either way. i think it would be nice not having to explicitly type null if you want an empty notification, it's a little less work, but also i wouldn't be opposed to being have to be explicit about it